include/aspect/geometry_model/box.h

```diff
@@ -105,6 +110,15 @@ namespace aspect
          */
         double depth(const Point<dim> &position) const override;
 
+        /**
+         * Function that uses the stored surface topography to calculate the depth
+         * including changes from mesh deformation. This will linearly interpolate
+         * between the two nearest surface points to get a depth for the given
+         * position.
+         */
+        double depth_including_mesh_deformation(const Point<dim> &position) const override;
```

Comment:
@gassmoeller suggests to just extend the existing `depth()` function by another parameter. I'm not sure that's the right approach because changing the interface of a `virtual` function requires changing it in all derived classes (some of which may just be out there in user land). I'm ok with introducing a new function in the base class that by default just throws an exception, and that individual geometry plugins can overload.

Reply:
I've kept the function as is, though for now it is only called in the visualization plugins.